core/src/main/scala/kafka/log/LogLoader.scala

```diff
@@ -191,7 +192,7 @@ class LogLoader(
     // Reload all snapshots into the ProducerStateManager cache, the intermediate ProducerStateManager used
     // during log recovery may have deleted some files without the LogLoader.producerStateManager instance witnessing the
     // deletion.
-    producerStateManager.removeStraySnapshots(segments.baseOffsets.toSeq)
+    producerStateManager.removeStraySnapshots(segments.baseOffsets.toSeq.map(x => Long.box(x)).asJava)
```

Comment:
Seems like we do two collection copies here. Maybe we can tweak things so that only one of them is needed (and when we convert the segments code, none are needed.

Reply:
Good catch! Avoided `Seq` conversion by using Collection as arg. We do not need these conversions when we convert LogSegment, which is planned in few weeks.